oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/editor/FulltextDocumentMaker.java

```diff
@@ -105,6 +108,16 @@ public FulltextDocumentMaker(@Nullable FulltextBinaryTextExtractor textExtractor
 
     protected abstract void indexNodeName(D doc, String value);
 
+    protected void logLargeStringProperties(String propertyName, String value) {
+        int logWarnStringSizeThreshold = DEFAULT_WARN_LOG_STRING_SIZE_THRESHOLD_VALUE;
+        if (System.getProperty(WARN_LOG_STRING_SIZE_THRESHOLD_KEY) != null && !System.getProperty(WARN_LOG_STRING_SIZE_THRESHOLD_KEY).isEmpty()) {
+            logWarnStringSizeThreshold = Integer.parseInt(System.getProperty(WARN_LOG_STRING_SIZE_THRESHOLD_KEY));
+        }
```

Comment:
This should probably be done in constructor to some `private final int logWarnStringSIzeThreshold` field.

Reply:
Done